### DISCUSSION TITLE
How do I get the image format?

### DISCUSSION THREAD

**Discussion by chaoxi24** [libvips/libvips#3515] [General]
How do I get a .jpg image format?

I have some png images on my computer that are incorrectly set to jpg files, which function do I need to use to get the true format of an image?

Sorry I'm new to this, I've spent hours trying to solve this problem, but I only got it through "interpretation"
((VipsInterpretation) VIPS_INTERPRETATION_sRGB)

**Comment by lovell** [comment]
Did you see `vips_foreign_find_load()`, which will inspect an image header/metadata to determine its format?

https://www.libvips.org/API/current/VipsForeignSave.html#vips-foreign-find-load
